Assets/Tayx/Graphy - Ultimate Stats Monitor/Scripts/Util/G_FloatString.cs

```diff
@@ -207,14 +206,18 @@ public static float ToFloat(this int i)
 
         #region Methods -> Private
 
-        //TODO: Replace this with a better algorithm.
-        private static int Pow(int f, int p)
-        {
-            for (int i = 1; i < p; i++)
-            {
-                f *= f;
+        // With the loop there is p operations, but with this method there is log2(p) operations
+        // source : https://en.wikipedia.org/wiki/Exponentiation_by_squaring
+        private static int Pow (int f, int p) {
+            if (0 == p) {
+                return 1;
+            } else if (1 == p) {
+                return f;
+            } else if (0 == p % 2) {
+                return Pow (f * f, p / 2);
+            } else {
+                return f * Pow (f * f, (p - 1) / 2);
```

Comment:
This is not tail recursive though.

Reply:
More importantly why do we have this Pow function and should we just remove it instead?

Reply:
What's the point of implementing a new algorithm if this Pow is called exactly once at initialization stage?

Reply:
There was a todo and I don't like todoes

Reply:
`Mathf.pow` should be a good alternative in this case.

Reply:
Wait, there was an for-loop implementation? `Mathf.Pow` is better for sure - it's probably using hardware `pow` implementation which is pseudo-"1 tick" implementation, not a loop one. Even if it's not using it, it's still faster/equal to log2 approach.

Reply:
Yes, there was a for loop. I don't really know why Mathf.Pow wasn't use, so I just improved the existing function, knowing it won't be the best approach.